prow/plank/controller.go

```diff
@@ -202,9 +228,11 @@ func (c *Controller) Sync() error {
 	}
 
 	// Share what we have for gathering metrics.
+	c.lock.Lock()
```

Comment:
Why wasn't this necessary in the past?

Reply:
I added this because I didn't properly understand Cole's comment about lock ordering, its not necessary and removed again